ibm/data_source_ibm_dl_gateway.go

```diff
@@ -0,0 +1,475 @@
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway BGP status",
+			},
+			dlCompletionNoticeRejectReason: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Reason for completion notice rejection",
+			},
+			dlCreatedAt: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The date and time resource was created",
+			},
+			dlCrn: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The CRN (Cloud Resource Name) of this gateway",
+			},
+			dlCrossConnectRouter: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Cross connect router",
+			},
+			dlDedicatedHostingID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Dedicated host id",
+			},
+			dlGlobal: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Gateways with global routing (true) can connect to networks outside their associated region",
+			},
+			dlLinkStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway link status",
+			},
+			dlLocationDisplayName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway location long name",
+			},
+			dlLocationName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway location",
+			},
+			dlMetered: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Metered billing option",
+			},
+
+			dlOperationalStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway operational status",
+			},
+			dlPort: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway port",
+			},
+			dlProviderAPIManaged: {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates whether gateway was created through a provider portal",
+			},
+			dlResourceGroup: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway resource group",
+			},
+			dlSpeedMbps: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Gateway speed in megabits per second",
+			},
+			dlType: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Gateway type",
+			},
+			dlVlan: {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "VLAN allocated for this gateway",
+			},
+		},
+	}
+}
+
+func dataSourceIBMDLGatewayVirtualConnectionsRead(d *schema.ResourceData, meta interface{}) error {
+	directLink, err := meta.(ClientSession).DirectlinkV1API()
+
+	if err != nil {
+		return err
+	}
+	listVcOptions := &directlinkapisv1.ListGatewayVirtualConnectionsOptions{}
+	dlGatewayId := d.Id()
+	listVcOptions.SetGatewayID(dlGatewayId)
+	listGatewayVirtualConnections, _, err := directLink.ListGatewayVirtualConnections(listVcOptions)
```

Comment:
listGatewayVirtualConnections, _, err : to listGatewayVirtualConnections, response, err :

	if err != nil {
		 return fmt.Errorf("Error while listing  XXX %s\n%s", err, response)
	}

We received an isssue yesterday for private dns error is not giving enough details